qiskit/visualization/pulse_v2/types.py

```diff
@@ -14,88 +14,189 @@
+ChartAxis.name.__doc__ = 'Name of chart.'
+ChartAxis.channels.__doc__ = 'Channels associated with chart.'
 
 
 ComplexColors = NamedTuple(
     'ComplexColors',
     [('real', str),
      ('imaginary', str)])
+ComplexColors.__doc__ = 'Data to represent a set of color codes for real and imaginary part.'
+ComplexColors.real.__doc__ = 'Color code of real part.'
+ComplexColors.imaginary.__doc__ = 'Color code of imaginary part.'
+
+
+ParsedInstruction = NamedTuple(
+    'ParsedWaveform',
+    [('xvals', np.ndarray),
+     ('yvals', np.ndarray),
+     ('meta', Dict[str, Any])]
+)
+ParsedInstruction.__doc__ = 'Data to represent a parsed pulse instruction for object generation.'
+ParsedInstruction.xvals.__doc__ = 'Numpy array of x axis data.'
+ParsedInstruction.yvals.__doc__ = 'Numpy array of y axis data.'
+ParsedInstruction.meta.__doc__ = 'Dictionary containing instruction details.'
+
+
+HorizontalAxis = NamedTuple(
+    'HorizontalAxis',
+    [('window', Tuple[int, int]),
+     ('axis_map', Dict[int, int]),
+     ('axis_break_pos', List[int]),
+     ('label', str)]
+)
+HorizontalAxis.__doc__ = "Data to represent configuration of horizontal axis."
+HorizontalAxis.window.__doc__ = "Left and right edge of graph."
+HorizontalAxis.axis_map.__doc__ = 'Mapping of apparent coordinate system and actual location.'
+HorizontalAxis.axis_break_pos.__doc__ = "Locations of axis break."
+HorizontalAxis.label.__doc__ = "Label of horizontal axis."
+
+
+class DrawingWaveform(str, Enum):
+    """
+    Waveform data type.
+
+    REAL: Assigned to objects that represent real part of waveform.
+    IMAG: Assigned to objects that represent imaginary part of waveform.
+    """
+    REAL = 'Waveform.Real'
+    IMAG = 'Waveform.Imag'
+
+
+class DrawingLabel(str, Enum):
+    """
+    Label data type.
+
+    PULSE_NAME: Assigned to objects that represent name of waveform.
+    CH_NAME: Assigned to objects that represent name of channel.
+    CH_SCALE: Assigned to objects that represent scaling factor of channel.
+    FRAME: Assigned to objects that represent value of frame.
+    SNAPSHOT: Assigned to objects that represent label of snapshot.
+    """
+    PULSE_NAME = 'Label.Pulse.Name'
+    PULSE_INFO = 'Label.Pulse.Info'
+    CH_NAME = 'Label.Channel.Name'
+    CH_INFO = 'Label.Channel.Info'
+    FRAME = 'Label.Frame.Value'
+    SNAPSHOT = 'Label.Snapshot'
+
+
+class DrawingSymbol(str, Enum):
+    """
+    Symbol data type.
+
+    FRAME: Assigned to objects that represent symbol of frame.
+    SNAPSHOT: Assigned to objects that represent symbol of snapshot.
+    """
+    FRAME = 'Symbol.Frame'
+    SNAPSHOT = 'Symbol.Snapshot'
+
+
+class DrawingLine(str, Enum):
+    """
+    Line data type.
+
+    BASELINE: Assigned to objects that represent zero line of channel.
+    BARRIER: Assigned to objects that represent barrier line.
+    """
+    BASELINE = 'Line.Baseline'
+    BARRIER = 'Line.Barrier'
+
+
+class AbstractCoordinate(str, Enum):
```

Comment:
Why do these need to inherit `str`?

Reply:
I need something that cannot be operated as a numerical object due to this logic:
https://github.com/nkanazawa1989/qiskit-terra/blob/567066adc0e600b923c6f8e34f738fe0a76fbd89/qiskit/visualization/pulse_v2/core.py#L674-L677

When the `AbstractCoordinate` is casted to float, this will cause an ValueError and another logic to bind float value is triggered.